hudi-flink/src/main/java/org/apache/hudi/sink/StreamWriteFunction.java

```diff
@@ -470,30 +470,36 @@ private void bufferRecord(HoodieRecord<?> value) {
     boolean flushBucket = bucket.detector.detect(item);
     boolean flushBuffer = this.tracer.trace(bucket.detector.lastRecordSize);
     if (flushBucket) {
-      flushBucket(bucket);
-      this.tracer.countDown(bucket.detector.totalSize);
-      bucket.reset();
+      boolean flush = flushBucket(bucket);
+      if (flush) {
+        this.tracer.countDown(bucket.detector.totalSize);
```

Comment:
flush => flushed

Reply:
fix

Reply:
Can we just inline this flag, make it more concise.

Reply:
ok